privacyidea/api/auth.py

```diff
@@ -59,7 +59,7 @@
 from privacyidea.lib.audit import getAudit
 from privacyidea.lib.auth import check_webui_user, ROLE
 from privacyidea.lib.user import User
-from privacyidea.lib.user import split_user
+from privacyidea.lib.user import split_user, log_used_user
```

Comment:
There are some endpoints that still write the secondary username to the audit log, like ``/token`` and ``/system``. Should we fix that as well? Maybe we can do that with a few lines in ``before_after.py``.

Reply:
I can not see, where this would happen in `/token`.

The token endpoints always use `user = get_user_from_param(request.all_data, required)` to determine the user object, which now returns the primary login attribute.

And here
https://github.com/privacyidea/privacyidea/blob/1082/secondary-login-attribute/privacyidea/api/before_after.py#L176
we call `get_user_from_param` which creates a `User` Object, that now will contain the primary login.

(Besides that I saw a duplicate which I will remove)

Reply:
I cleaned up some ``get_user_from_param`` in the token endpoint. This should totally be fine!
f64a4c12

Reply:
:+1: